framework/src/postprocessors/FindValueOnLine.C

```diff
@@ -0,0 +1,121 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "FindValueOnLine.h"
+#include "MooseMesh.h"
+
+template<>
+InputParameters validParams<FindValueOnLine>()
+{
+  InputParameters params = validParams<GeneralPostprocessor>();
+  params.addClassDescription("Find a specific target value along a sampling line. The variable values along the line should change monotonically. The target value is searched using a bisection algorithm.");
+  params.addParam<Point>("start_point", "Start point of the sampling line.");
+  params.addParam<Point>("end_point", "End point of the sampling line.");
+  params.addParam<Real>("target", "Target value to locate.");
+  params.addParam<unsigned int>("depth", 30, "Maximum number of bisections to perform.");
+  params.addParam<Real>("tol", 1e-10, "Stop search if a value within a symmetric interval of this with around the target is found.");
+  params.addCoupledVar("v", "Variable to inspect");
+  return params;
+}
+
+FindValueOnLine::FindValueOnLine(const InputParameters & parameters) :
+    GeneralPostprocessor(parameters),
+    Coupleable(this, false),
+    _start_point(getParam<Point>("start_point")),
+    _end_point(getParam<Point>("end_point")),
+    _length((_end_point - _start_point).norm()),
+    _target(getParam<Real>("target")),
+    _depth(getParam<unsigned int>("depth")),
+    _tol(getParam<Real>("tol") / 2.0),
+    _coupled_var(getVar("v", 0)),
+    _position(0.0),
+    _mesh(_subproblem.mesh()),
+    _point_vec(1)
+{
+}
+
+void
+FindValueOnLine::initialize()
+{
+  // We do this here just in case it's been destroyed and recreated becaue of mesh adaptivity.
+  _pl = _mesh.getMesh().sub_point_locator();
+}
+
+void
+FindValueOnLine::execute()
+{
+  Real s;
+  Real s_left = 0.0;
+  Real left = getValueAtPoint(_start_point);
+  Real s_right = 1.0;
+  Real right = getValueAtPoint(_end_point);
+
+  for (unsigned i = 0; i < _depth; ++i)
+  {
+    // find midpoint
+    s = (s_left + s_right) / 2.0;
+    Point p = s * (_end_point - _start_point) + _start_point;
+
+    // sample value
+    Real value = getValueAtPoint(p);
+
+    // have we hit the target value yet?
+    if (value > _target - _tol && value < _target + _tol)
```

Comment:
@bwspenc was the original implementor. He may have a reason, otherwise I propose that we adjust these methods appropriately and use them throughout the framework more.